### ISSUE TITLE
[require-jsdoc] False positive for exported overload functions

### ISSUE THREAD

**Issue by manuth** [gajus/eslint-plugin-jsdoc#645]
## Expected behavior
`TSDeclareFunction`s should report no error even if they have a prepended `export`.

## Actual behavior
An error is reported sayin that the jsdoc-comment is missing.

## ESLint Config
```js
{
    rules: {
        "jsdoc/require-jsdoc": [
            "warn",
            {
                require: {
                    FunctionDeclaration: true
                },
                contexts: [
                    "TSDeclareFunction"
                ]
            }
        ]
    }
}
```

## ESLint sample
```ts
/**
 * This is a text.
 */
function x(); // works great

/**
 * This is a text.
 */
function x(text: string) { }

/**
 * This is a text.
 */
export function a(); // Reports an error

/**
 * This is a text.
 */
export function a(text: string)
{ }
```

## Environment

- Node version: v12.18.3
- ESLint version v7.10.0
- `eslint-plugin-jsdoc` version: 30.6.2

**Comment by manuth** [comment]
Awesome, thank you so much - that was fast!
